src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,300 @@
+            padding_value=padding_value,
+            **kwargs,
+        )
+        self.n_bars = n_bars
+        self.sampling_rate = sampling_rate
+        self.use_mel = use_mel
+        self.padding_value = padding_value
+        self.fft_window_size = fft_window_size
+        self.hop_length = hop_length
+        self.frequency_min = frequency_min
+        self.nb_mel_filters = nb_mel_filters
+
+    def log_mel_spectogram(self, sequence):
+        """Generates MelSpectrogram then applies log base e."""
+
+        mel_fb = get_mel_filter_banks(
+            nb_frequency_bins=(self.fft_window_size // 2) + 1,
+            nb_mel_filters=self.nb_mel_filters,
+            frequency_min=self.frequency_min,
+            frequency_max=float(self.sampling_rate // 2),
+            sample_rate=self.sampling_rate,
+            norm=None,
+            mel_scale="htk",
+        ).astype(np.float32)
+
+        spectrogram = []
+        for seq in sequence:
+            window = np.hanning(self.fft_window_size + 1)[:-1]
+            framed_audio = fram_wave(seq, self.hop_length, self.fft_window_size)
+            spec = stft(framed_audio, window, fft_window_size=self.fft_window_size)
+            spec = np.abs(spec) ** 2.0
+            spectrogram.append(spec)
+
+        spec_shape = spec.shape
+        spectrogram = torch.Tensor(spectrogram).view(-1, *spec_shape)
+        log_melspec = (
+            torch.matmul(spectrogram.transpose(-1, -2), torch.from_numpy(mel_fb))
+            .transpose(-1, -2)
+            .clamp(min=1e-6)
+            .log()
+        )
+
+        return log_melspec
+
+    def extract_rhythm(self, raw_audio):
+        """
+        This algorithm(`RhythmExtractor2013`) extracts the beat positions and estimates their confidence as well as
+        tempo in bpm for an audio signal. For more information please visit
+        https://essentia.upf.edu/reference/std_RhythmExtractor2013.html .
+        """
+        essentia_tracker = essentia.standard.RhythmExtractor2013(method="multifeature")
+        bpm, beat_times, confidence, estimates, essentia_beat_intervals = essentia_tracker(raw_audio)
+
+        return bpm, beat_times, confidence, estimates, essentia_beat_intervals
+
+    def interpolate_beat_times(self, beat_times, steps_per_beat, extend=False):
+        beat_times_function = scipy.interpolate.interp1d(
+            np.arange(beat_times.size),
+            beat_times,
+            bounds_error=False,
+            fill_value="extrapolate",
+        )
+        if extend:
+            beat_steps_8th = beat_times_function(np.linspace(0, beat_times.size, beat_times.size * steps_per_beat + 1))
+        else:
+            beat_steps_8th = beat_times_function(
+                np.linspace(0, beat_times.size - 1, beat_times.size * steps_per_beat - 1)
+            )
+        return beat_steps_8th
+
+    def extrapolate_beat_times(self, beat_times, n_extend=1):
+        beat_times_function = scipy.interpolate.interp1d(
```

Comment:
Is this equivalent to `numpy.interp`? https://numpy.org/doc/stable/reference/generated/numpy.interp.html Happy to keep as `scipy.interpolate` if not, but just wondering if we can reduce our dependency list by one

Reply:
Actually I tried to use numpy.interp instead of scipy, but it seemed that those two gave different outputs.

Reply:
Sure - good for me to stick with scipy in this case